extensions/planetscale/src/create-branch.tsx

```diff
@@ -0,0 +1,98 @@
+import { FormValidation, useForm } from "@raycast/utils";
+import { Action, ActionPanel, Form } from "@raycast/api";
+import { isEmpty, sample } from "lodash";
+import { useState } from "react";
+import { View } from "./lib/oauth/view";
+import { FormDatabaseDropdown } from "./lib/components/form-database-dropdown";
+import { useBranches } from "./lib/hooks/use-branches";
+import { getBranchIcon } from "./lib/icons";
+
+interface CreateBranchForm {
+  name: string;
+  parent: string;
+  database: string;
+}
+
+const placeholderBranchNames = [
+  "321-replace-postgres-with-mysql",
+  "911-remove-elephant-database",
+  "153-give-steve-the-editor-a-raise",
+  "107-create-indices-aaron-suggested",
+];
+
+export function CreateBranch(
+  props:
+    | {
+        organization: string;
+        database: string;
+        branch: string;
+      }
+    | object,
+) {
+  const [placeholderBranchName] = useState(() => sample(placeholderBranchNames));
+
+  const { itemProps, values, handleSubmit, reset } = useForm<CreateBranchForm>({
+    validation: {
+      name: FormValidation.Required,
+      parent: FormValidation.Required,
+      database: FormValidation.Required,
+    },
+    initialValues: {
+      name: "",
+      parent: "branch" in props ? props.branch : "",
+      database: "organization" in props ? `${props.organization}-${props.database}` : "",
+    },
+    async onSubmit(values) {
+      const [organization, database] = values.database.split("-");
+
+      await createBranch({
+        name: values.name,
+        parent: values.parent,
+        database,
+        organization,
+      });
+
+      reset({ name: "", database: values.database, parent: values.parent });
+    },
+  });
+
+  const [organization, database] = values.database.split("-");
+  const { branches, branchesLoading, createBranch } = useBranches({ organization, database });
+
+  return (
+    <Form
+      enableDrafts={!props}
```

Comment:
If you enable the draft feature, you need to pass the form's draft values (that you get from the [LaunchProps](https://developers.raycast.com/information/lifecycle)) as `initialValues` otherwise, there's no point in using it.

Reply:
Fixed!

Uploading CleanShot 2024-01-30 at 17.55.37.mp4…